test/integration/hcpvaultsecretsapp_integration_test.go

```diff
@@ -0,0 +1,405 @@
+			appName := fmt.Sprintf("vso-test-%x", sum[:8])
+
+			obj := &secretsv1beta1.HCPVaultSecretsApp{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      appName,
+					Namespace: outputs.K8sNamespace,
+				},
+				Spec: secretsv1beta1.HCPVaultSecretsAppSpec{
+					AppName:      appName,
+					RefreshAfter: "3s",
+					Destination: secretsv1beta1.Destination{
+						Create: true,
+						Name:   appName,
+					},
+					RolloutRestartTargets: []secretsv1beta1.RolloutRestartTarget{
+						{
+							Kind: "Deployment",
+							Name: appName,
+						},
+					},
+				},
+			}
+			require.NoError(t, crdClient.Create(ctx, obj))
+			objsCreated = append(objsCreated, obj)
+
+			if len(obj.Spec.RolloutRestartTargets) > 0 {
+				depObj := &appsv1.Deployment{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      appName,
+						Namespace: outputs.K8sNamespace,
+						Labels: map[string]string{
+							"test": appName,
+						},
+					},
+					Spec: appsv1.DeploymentSpec{
+						Selector: &v1.LabelSelector{
+							MatchLabels: map[string]string{
+								"test": appName,
+							},
+						},
+						Replicas: pointer.Int32(3),
+						Template: corev1.PodTemplateSpec{
+							ObjectMeta: v1.ObjectMeta{
+								Labels: map[string]string{
+									"test": appName,
+								},
+							},
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name:  appName,
+										Image: "busybox:latest",
+										Command: []string{
+											"sh", "-c", "while : ; do sleep 10; done",
+										},
+									},
+								},
+								TerminationGracePeriodSeconds: pointer.Int64(2),
+							},
+						},
+						Strategy: appsv1.DeploymentStrategy{
+							RollingUpdate: &appsv1.RollingUpdateDeployment{
+								MaxUnavailable: &intstr.IntOrString{
+									Type:   intstr.String,
+									StrVal: "25%",
+								},
+							},
+						},
+					},
+				}
+				require.NoError(t, crdClient.Create(ctx, depObj))
+				objsCreated = append(objsCreated, depObj)
+			}
+
+			t.Cleanup(func() {
+				if !skipCleanup {
+					for _, obj := range objsCreated {
+						assert.NoError(t, crdClient.Delete(ctx, obj))
+					}
+
+					deleteHVSApp(t, ctx, hvsClient, hcpOrganizationID, hcpProjectID, appName)
+				}
+			})
+			_, err := hvsClient.CreateApp(
+				hvsclient.NewCreateAppParams().
+					WithLocationOrganizationID(hcpOrganizationID).
+					WithLocationProjectID(hcpProjectID).
+					WithContext(ctx).
+					WithBody(hvsclient.CreateAppBody{
+						Description: fmt.Sprintf("VSO test %s", t.Name()),
+						Name:        appName,
+					}), nil)
+			require.NoError(t, err, "failed to create app %q", appName)
+
+			require.NoError(t, createHVSSecrets(t, appName, tt.expectedData))
+
+			_, err = awaitSecretSynced(t, ctx, crdClient, obj, tt.expectedData)
+			if assert.NoError(t, err) {
+				if len(tt.updateData) > 0 {
+					require.NoError(t, createHVSSecrets(t, appName, tt.updateData))
+					_, err = awaitSecretSynced(t, ctx, crdClient, obj, tt.updateData)
```

Comment:
Should this `err` value be checked?

Reply:
Good catch! Fixed with 4298d90e8471bbac5c59f4ff45f30432c4eb762b